### PR TITLE
MINOR: Add equals/hashcode to DeprecatedRequestRateKey

### PR DESCRIPTION
**Pull request by mimaison**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by mimaison** [comment]
Following https://github.com/apache/kafka/pull/17015#issuecomment-2338011367
cc @lucasbru @chia7712

File: server/src/main/java/org/apache/kafka/network/metrics/RequestMetrics.java [hunk 1]
```diff
@@ -214,5 +215,20 @@ private DeprecatedRequestRateKey(short version, ClientInformation clientInformat
...
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            DeprecatedRequestRateKey that = (DeprecatedRequestRateKey) o;
+            return version == that.version && Objects.equals(clientInformation, that.clientInformation);
+        }
+
+        @Override
+        public int hashCode() {
+            int result = version;
```
Comment (chia7712) [inline server/src/main/java/org/apache/kafka/network/metrics/RequestMetrics.java]: How about using `Objects.hash` to simplify the code?

**Review by chia7712 — COMMENTED** [review]
@mimaison thanks for this quick fix!

File: server/src/main/java/org/apache/kafka/network/metrics/RequestMetrics.java (hunk 1 above)
Comment (mimaison) [reply server/src/main/java/org/apache/kafka/network/metrics/RequestMetrics.java]: Sure, done

**Review by chia7712 — APPROVED** [review]
LGTM

**Review by lucasbru — APPROVED** [review]
Thanks for the quick fix, LGTM!

**Pull request merged**